src/tp/tp_segmenter.cpp

```
@@ -42,31 +41,29 @@ TpSegmenter::TpSegmenter(const TpConfig& config)
-        Message tp_message = message;
-        if (payload.size() > 1000) {  // Threshold for when to use TP even for single segments
-            tp_message.set_message_type(add_tp_flag(message.get_message_type()));
-        }
-        platform::ByteBuffer message_data = tp_message.serialize();
+        // Payload fits in one non-TP SOME/IP message: no TP-Flag, no TP header.
+        platform::ByteBuffer message_data = message.serialize();
 
         TpSegment segment;
         segment.header.message_type = TpMessageType::SINGLE_MESSAGE;
         segment.header.message_length = static_cast<uint32_t>(payload.size());
         segment.header.segment_offset = 0;
```

Cppcheck (2.21.0)

[style] The function 'subview' is never used.

(unusedFunction)